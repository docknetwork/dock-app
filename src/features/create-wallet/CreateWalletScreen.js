import React, { useState } from 'react';
import {
  Image,
  Button,
  LoadingButton,
  Header,
  Footer,
  Content,
  ScreenContainer,
  Box,
  Theme,
} from '../../design-system';
import styled, {ThemeProvider} from 'styled-components/native';

import SplashLogo from '../../assets/splash-logo.png';
import {navigate} from '../../core/navigation';
import {Routes} from '../../core/routes';

export function CreateWalletScreen() {
  return (
    <ScreenContainer testID="createWalletScreen">
      <Header></Header>
      <Content>
        <Box flex={1} justifyContent="center" row marginTop="50%">
          <Image
            source={SplashLogo}
            style={{
              width: '57%',
            }}
            resizeMode="contain"
          />
        </Box>
      </Content>
      <Footer marginBottom={114} marginLeft={26} marginRight={26}>
        <LoadingButton
          full
          testID="createWalletButton"
          onPress={() => navigate(Routes.CREATE_WALLET_PASSCODE)}>
          Create a new wallet
        </LoadingButton>
      </Footer>
    </ScreenContainer>
  );
}
