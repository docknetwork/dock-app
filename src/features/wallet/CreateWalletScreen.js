import React from 'react';
import SplashLogo from '../../assets/splash-logo.png';
import {navigate} from '../../core/navigation';
import {Routes} from '../../core/routes';
import {
  Box,
  Content,
  Footer,
  Header,
  Image,
  LoadingButton,
  ScreenContainer,
} from '../../design-system';
import {translate} from '../../locales';
import {BuildIdentifier} from '../app/BuildIdentifier';

export function CreateWalletScreen() {
  return (
    <ScreenContainer testID="createWalletScreen">
      <Header />
      <Content>
        <Box flex={1} justifyContent="center" row marginTop="50%">
          <Image
            source={SplashLogo}
            style={{
              width: '57%',
            }}
            resizeMode="contain"
          />
        </Box>
      </Content>
      <Footer marginBottom={10} marginLeft={26} marginRight={26}>
        <LoadingButton
          full
          testID="createWalletButton"
          onPress={() => navigate(Routes.CREATE_WALLET_PASSCODE)}>
          {translate('create_wallet.create_new')}
        </LoadingButton>
        <LoadingButton
          full
          variant="ghost"
          mt={4}
          testID="createWalletButton"
          onPress={() => navigate(Routes.WALLET_IMPORT_BACKUP)}>
          {translate('create_wallet.import_existing')}
        </LoadingButton>
        <BuildIdentifier />
      </Footer>
    </ScreenContainer>
  );
}
