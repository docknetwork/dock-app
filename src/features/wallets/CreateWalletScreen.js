import {
  Form,
  H1,
  Input,
  Item,
  Label,
  Header,
  Footer,
  Content,
  Button,
  Text,
  Container,
  View,
  Spinner,
} from 'native-base';
import React, {useState} from 'react';
import {InteractionManager} from 'react-native';
import {useDispatch} from 'react-redux';
import {Colors} from '../../theme/colors';
import {walletsOperations} from './wallets-slice';

export function CreateWalletScreen({navigation}) {
  const dispatch = useDispatch();
  const [form, setForm] = useState({
    walletName: null,
    password: null,
    confirmPassword: null,
  });
  const [error, setError] = useState();
  const [loading, setLoading] = useState();

  const handleFormChange = key => value =>
    setForm({
      ...form,
      [key]: value,
    });

  const handleCreateWallet = () => {
    if (!form.password || !form.walletName) {
      setError('Please enter the required fields');
      return;
    }

    if (form.password !== form.confirmPassword) {
      setError("Password confirmation doesn't match");
      return;
    }

    setLoading(true);

    InteractionManager.runAfterInteractions(() => {
      dispatch(
        walletsOperations.createWallet({
          password: form.password,
          walletName: form.walletName,
        }),
      ).then(() => {
        setLoading(false);
      });
    });
  };

  return (
    <Container testID="createWalletScreen">
      <Header style={{alignItems: 'center', marginTop: 10}}>
        <H1 style={{color: '#fff'}}>Create Wallet</H1>
      </Header>
      <Content>
        <Form>
          <Item floatingLabel>
            <Label>Wallet name</Label>
            <Input
              onChangeText={handleFormChange('walletName')}
              value={form.walletName}
            />
          </Item>
          <Item floatingLabel>
            <Label>Password</Label>
            <Input
              secureTextEntry={true}
              onChangeText={handleFormChange('password')}
              value={form.password}
            />
          </Item>
          <Item floatingLabel last>
            <Label>Confirm Password</Label>
            <Input
              secureTextEntry={true}
              onChangeText={handleFormChange('confirmPassword')}
              value={form.confirmPassword}
            />
          </Item>
        </Form>
        <View style={{padding: 20}}>
          {<Text style={{color: Colors.red}}>{error}</Text>}
        </View>
      </Content>
      <Footer style={{backgroundColor: '#fff', padding: 8}}>
        <Button
          testID="createWalletBtn"
          full
          style={{width: '100%'}}
          primary
          disabled={loading}
          onPress={handleCreateWallet}>
          {loading ? (
            <Spinner color="#fff" size={15} />
          ) : (
            <Text style={{color: '#fff'}}>Create Wallet</Text>
          )}
        </Button>
      </Footer>
    </Container>
  );
}
