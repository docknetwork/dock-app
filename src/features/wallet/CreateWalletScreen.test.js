import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {CreateWalletScreen} from './CreateWalletScreen';

const mockStore = configureMockStore();

describe('CreateWalletScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<CreateWalletScreen />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
