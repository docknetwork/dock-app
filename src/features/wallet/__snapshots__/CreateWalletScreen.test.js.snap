// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWalletScreen should render correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#18181B",
      "flex": 1,
    }
  }
  testID="createWalletScreen"
>
  <Styled(View) />
  <Styled(ScrollView)>
    <Box
      flex={1}
      justifyContent="center"
      marginTop="50%"
      row={true}
    >
      <Styled(Image)
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/splash-logo.png",
          }
        }
        style={
          Object {
            "width": "57%",
          }
        }
      />
    </Box>
  </Styled(ScrollView)>
  <Styled(Box)
    marginBottom={10}
    marginLeft={26}
    marginRight={26}
  >
    <LoadingButton
      accessibilityLabel="createWalletButton"
      full={true}
      onPress={[Function]}
      testID="createWalletButton"
    >
      Create a new wallet
    </LoadingButton>
    <LoadingButton
      accessibilityLabel="import_existing"
      full={true}
      mt={4}
      onPress={[Function]}
      testID="import_existing"
      variant="ghost"
    >
      I already have a wallet
    </LoadingButton>
    <BuildIdentifier />
  </Styled(Box)>
</RNCSafeAreaView>
`;
